scene/3d/mesh_instance_3d.cpp

```diff
@@ -493,6 +493,93 @@ void MeshInstance3D::create_debug_tangents() {
 	}
 }
 
+Ref<ArrayMesh> MeshInstance3D::create_mesh_from_blend_shape_mix() {
+	Ref<ArrayMesh> new_baked_mesh;
+	new_baked_mesh.instantiate();
+	Ref<ArrayMesh> source_mesh = get_mesh();
+	ERR_FAIL_NULL_V_MSG(source_mesh, new_baked_mesh, "No valid source mesh with blend shapes");
+
+	for (int surface_index = 0; surface_index < source_mesh->get_surface_count(); surface_index++) {
+		const Array &source_mesh_arrays = source_mesh->surface_get_arrays(surface_index);
+
+		ERR_FAIL_COND_V(source_mesh_arrays.size() != RS::ARRAY_MAX, new_baked_mesh);
+
+		const Vector<Vector3> &source_mesh_vertex_array = source_mesh_arrays[Mesh::ARRAY_VERTEX];
+		const Vector<Vector3> &source_mesh_normal_array = source_mesh_arrays[Mesh::ARRAY_NORMAL];
+		const Vector<float> &source_mesh_tangent_array = source_mesh_arrays[Mesh::ARRAY_TANGENT];
+
+		Array new_mesh_arrays;
+		new_mesh_arrays.resize(Mesh::ARRAY_MAX);
+		for (int i = 0; i < source_mesh_arrays.size(); i++) {
+			if (i == Mesh::ARRAY_VERTEX || i == Mesh::ARRAY_NORMAL || i == Mesh::ARRAY_TANGENT) {
+				continue;
+			}
+			new_mesh_arrays[i] = source_mesh_arrays[i];
+		}
+
+		Vector<Vector3> lerped_vertex_array = source_mesh_vertex_array;
+		Vector<Vector3> lerped_normal_array = source_mesh_normal_array;
+		Vector<float> lerped_tangent_array = source_mesh_tangent_array;
+
+		const Array &blendshapes_mesh_arrays = source_mesh->surface_get_blend_shape_arrays(surface_index);
+		int blend_shape_count = source_mesh->get_blend_shape_count();
+		ERR_FAIL_COND_V(blendshapes_mesh_arrays.size() != blend_shape_count, new_baked_mesh);
+
+		for (int blendshape_index = 0; blendshape_index < blend_shape_count; blendshape_index++) {
+			float blend_weight = get_blend_shape_value(blendshape_index);
+			if (abs(blend_weight) > 0.0001) {
```

Comment:
blend weights can be negative. if the idea is to reduce performance, perhaps consider `is_zero_approx` or `is_equal_approx`

Reply:
Sure I can change it to use the Math class but this use of `abs(blend_weight) > 0.0001` is more or less a straight copy from the skeleton.glsl shader.

Reply:
I apologize. I missed the use of `abs()` in your code so I think it is fine.

It should use the same magic number as skeleton.glsl, and if that is indeed 0.0001 then your code is correct, so disregard my comment.